Drift/Private/DriftBase.cpp

```diff
@@ -1188,6 +1188,20 @@ void FDriftBase::HandleFriendEventMessage(const FMessageQueueEntry& message)
 
         onFriendRemoved.Broadcast(message.sender_id);
     }
+    else if (eventName == TEXT("friend_request"))
+    {
+        UE_LOG(LogDriftMessages, Verbose, TEXT("Player %d wants to be friends with us. Awwww..."), message.sender_id);
+        auto token = message.payload.FindField(TEXT("token"));
+        if (!token.IsString())
+        {
+            UE_LOG(LogDriftMessages, Error, TEXT("Missing or invalid friend invite token"));
+        }
+        onFriendRequestReceived.Broadcast(message.sender_id, token.ToString());
+    }
+    else
+    {
+        UE_LOG(LogDriftMessages, Warning, TEXT("Unknown event '%s' not handled"), *event.ToString());
```

Comment:
I'd prefer this not be a warning, or logged at all, except possibly in development builds, since when we upgrade the backend with new messages, we don't want old clients to start spitting out warnings.